provider/README.md

```diff
@@ -9,7 +9,9 @@ Searches for a provider in this order:
 1. `providerArg`, passed in via CLI
 2. Environment variable `SUMMON_PROVIDER`
 3. Executable in `/usr/local/lib/summon`
-(or `%ProgramW6432%\Cyberark Conjur\Summon\Providers` on Windows).
+   (or `%ProgramW6432%\Cyberark Conjur\Summon\Providers` on Windows).
+4. If all of the above do net exist: look for Executable in 
```

Comment:
So it seems it checks if the directory`/usr/local/lib/summon` exists, and not if there is an executable in the directory.  For example if I move the provider executable from `/usr/local/lib/summon` to `./Provider` it will not find the new location unless I delete the `summon` directory. Can you add something like
If all of the above are not set and if the directory `/usr/local/lib/summon` does not exist, look for the Executable in`<path_to_summon_excutable>\Providers` (aka 'portable mode')

Reply:
@rpothier you are right, but the original idea was to have no system directory in portable mode. I thought of the portable mode as something that does not change any system wide configuration or directory. It was therefore, in my current implementation, never intended to handle multiple directories at the same time (i.e. system and local/portable one). 

So here's the question: do we want to support system and local provider dirs in parallel? Then I can not handle it in the `getDefaultPath` function but need to search providers in several directories when [resolving the actual configured provider](https://github.com/cyberark/summon/blob/35d826f04626b35c522c594f7129da611e2a760b/provider/provider.go#L111).

What do you think: exclusive either/or of the directories or search within all of them? We'd also need to consider priorities if one provider is available in both dirs. Things get a little bit more complicated in case of handling multiple dirs ;-)

Reply:
... or i could check for empty system provider dir and resort to local dir in that case only. But this all seems to complicate thinks 🤷🏽‍♂️

Reply:
Hi, I think the way it is currently implemented is good, I was just commenting that we should document the behavior.
So if someone is migrating from system to local they will know to delete the system directory.

Reply:
I've update the provider readme, hopefully making it more clear how the provider search works and how one can migrate from system to local provider dir